### ISSUE TITLE
AutoML Early Stopping: Additional Validation Set

### ISSUE THREAD

**Issue by jeremyliweishih** [alteryx/evalml#280]
@dsherry proposed in #241 to utilize a separate validation set to test for early stopping. This could also be potentially extended to replaced cross validation in general. 

- The goal of early stopping is to detect when the pipeline starts to overfit on the training data; without a validation split, we won't see that the pipeline's predictive accuracy on data with a similar signal to the training data (but with different noise) has begun to drop. So I suggest we create a validation split in the fit method, and add guard rails to make sure the user has passed enough data to do that. (We could do CV for small training sets, but that bears further thought)```

- In the case when the data is large enough, and the training and validation splits are super i.i.d., then theoretically, CV would just slow down the evaluation without improving accuracy. But for smaller datasets, CV will protect against overfitting, to a point.```

**Comment by dsherry** [comment]
Closing, @jeremyliweishih let's chat if you feel we should keep tracking something for this.
